Tasks/Common/codecoverage-tools/jacoco/jacoco.maven.ccenabler.ts

```diff
@@ -0,0 +1,186 @@
+/// <reference path="../../../../definitions/Q.d.ts" />
+/// <reference path="../../../../definitions/string.d.ts" />
+/// <reference path="../../../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../../../definitions/node.d.ts" />
+
+import * as util from "../utilities";
+import * as tl from "vsts-task-lib/task";
+import * as ccc from "../codecoverageconstants";
+import * as cc from "../codecoverageenabler";
+import * as str from "string";
+import * as path from "path";
+import * as os from "os";
+import * as Q from "q";
+
+export class JacocoMavenCodeCoverageEnabler extends cc.JacocoCodeCoverageEnabler {
+
+    excludeFilter: string[];
+    includeFilter: string[];
+    reportDir: string;
+    sourceDirs: string;
+    classDirs: string;
+    reportBuildFile: string;
+
+    // -----------------------------------------------------
+    // Enable code coverage for Jacoco Maven Builds
+    // - enableCodeCoverage: CodeCoverageProperties  - ccProps
+    // -----------------------------------------------------    
+    public enableCodeCoverage(ccProps: { [name: string]: string }): Q.Promise<boolean> {
+        let _this = this;
+
+        _this.buildFile = ccProps["buildfile"];
+        _this.reportDir = ccProps["reportdirectory"];
+        _this.sourceDirs = ccProps["sourcedirectories"];
+        _this.classDirs = ccProps["classfilesdirectories"];
+        _this.reportBuildFile = ccProps["reportbuildfile"];
+
+        let classFilter = ccProps["classfilter"];
+        let filter = _this.extractFilters(classFilter);
+        _this.excludeFilter = _this.applyFilterPattern(filter.excludeFilter);
+        _this.includeFilter = _this.applyFilterPattern(filter.includeFilter);
+
+        return util.readXmlFileAsJson(_this.buildFile)
+            .then(function (resp) {
+                _this.addCodeCoverageData(resp);
+            })
+            .thenResolve(true);
```

Comment:
shouldnt all of the enablecodecoverage promises have a .catch to handle the error conditions?